src/test/server/store-test.js

```diff
@@ -1,9 +1,203 @@
-const UserStore = require('../../../_compiled_/stores/user-store').default;
+const assert = require('assert');
+
+const usersData  = require('../../test/data/users.json');
+const UserAction = require('../../../_compiled_/actions/user-action').default;
+const UserInfo   = require('../../../_compiled_/stores/user-info').default;
+const UserStore  = require('../../../_compiled_/stores/user-store').default;
+const EVENT_TYPE = require('../../../_compiled_/stores/event-type').default;
+
+const TEST_USER = new UserInfo(
+    {
+        name:    'Ervin New',
+        phone:   '89500418181',
+        email:   'vika.chernookaya@mail.ru',
+        website: 'github.com'
+    }
+);
+
+const TEST_USER_EXISTING = new UserInfo(
+    {
+        name:    'Ervin Howell',
+        phone:   '0418181',
+        email:   'ervin@mail.ru',
+        website: 'github.com'
+    }
+);
+
+const SEARCH_USER_INFO = new UserInfo({ name: 'Ervin' });
+
+const SEARCH_NON_EXIST_USER_INFO = new UserInfo({ name: 'Ivan' });
+
+const UPDATE_USER = new UserInfo(
+    {
+        id:      2,
+        name:    'Vika',
+        phone:   '555',
+        email:   'ervin@mail.ru',
+        website: 'github.com'
+    }
+);
+
+const SEARCH_FIELDS = {
+    name:    true,
+    phone:   false,
+    email:   false,
+    website: false
+};
+
+function subEvent (name, raiseAction) {
+    return new Promise(resolve => {
+        UserStore.on(name, (...args) => {
+            resolve(args);
+        });
+        raiseAction();
+    });
+}
+
+function _getInitialUserList () {
+    return usersData.map((user) => {
+        return new UserInfo(
+            {
+                id:      user.id,
+                name:    user.name,
+                phone:   user.phone,
+                email:   user.email,
+                website: user.website
+            },
+            user.address,
+            user.company
+        );
+    });
+}
 
 describe('UserStore', () => {
+    afterEach(() => {
+        UserStore._users        = _getInitialUserList();
+        UserStore._searchedInfo = null;
+        UserStore.removeAllListeners(EVENT_TYPE.userAdded);
+        UserStore.removeAllListeners(EVENT_TYPE.addingFailed);
+        UserStore.removeAllListeners(EVENT_TYPE.change);
+        UserStore.removeAllListeners(EVENT_TYPE.usersFound);
+    });
+
     describe('.getUsers()', () => {
         it('Should return a list of users', () => {
-            console.log(UserStore.getUsers());
+            const users = _getInitialUserList();
+
+            assert.deepEqual(UserStore.getUsers(), users);
+        });
+    });
+
+    describe('Actions', () => {
+        describe('addNewUser', () => {
+            it('Should emit userAdded event', async () => {
+                let newUser = TEST_USER;
```

Comment:
When you try to copy an object you copy only a link to this object.
And in this case if you modify a new obkect (`newUser `) you modify the origin object (`TEST_USER`). But you should not.
You need to make a deep copy 
Use https://lodash.com/docs/4.17.15#cloneDeep